storage/range.go

```diff
@@ -387,11 +387,16 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 	return nil
 }
 
-// verifyLeaderLease checks whether the requesting replica (by raft
-// node ID) holds the leader lease covering the specified timestamp.
-func (r *Range) verifyLeaderLease(originRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) bool {
-	l := r.getLease()
-	return originRaftNodeID == l.RaftNodeID && timestamp.Less(l.Expiration)
+// verifyLeaderLease checks whether the requesting replica (by raft node ID)
+// holds the leader lease covering the specified timestamp.
+func verifyLeaderLease(lease *proto.Lease, replicaRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) (bool, bool) {
```

Comment:
`func (r *Range) HasLiveLeaderLease(timestamp proto.Timestamp) bool` can't work because we often need to verify whether a specified replica has the lease at a time.  When a raft command is applied all replicas verify that the origin replica held the lease at the time the command was proposed.  We still need to be able to pass in the node id. 

"Live" is confusing here since we're passing in a timestamp and that timestamp will often be in the past.

I would try splitting this function into `verifyLeaderLease(*proto.Lease, proto.RaftNodeID, proto.Timestamp) bool` (returning the first result of this function) and `isLeaseAvailable(*proto.Lease, proto.Timestamp)` (returning the second).